src/modules/fw_att_control/ecl_pitch_controller.cpp

```diff
@@ -102,11 +102,61 @@ float ECL_PitchController::control_bodyrate(const float dt, const ECL_ControlDat
 		_integrator = math::constrain(_integrator + id * _k_i, -_integrator_max, _integrator_max);
 	}
 
+	// Calculate the contribution of the rate gyro to the actuator demand and high pass filter
+	const float p_term = _rate_error * _k_p * ctl_data.scaler * ctl_data.scaler;
+	float filter_input;
+	if (_k_ff > 0.0f) {
+		// If feed forward is being used then we can monitor the rate error term
+		// because the feed forward should be doing most of the work
+		filter_input = p_term;
+	} else {
+		// Some users will use a zero feed forward gain and rely on the rate error term
+		// to do all the work, resuling in large rate error tranisents. In this scenario
+		// it is better to monitor the gyro feedback contribution only
+		filter_input = ctl_data.body_y_rate * _k_p * ctl_data.scaler * ctl_data.scaler;
+	}
+	const float filt_tconst = _tc / 2.0f;
+	const float filt_coef = fminf(dt, filt_tconst) / filt_tconst;
+	_gyro_contribution_hpf = (1.0f - filt_coef) * _gyro_contribution_hpf + filt_coef * filter_input;
+	const float gyro_term_hpf = filter_input - _gyro_contribution_hpf;
+
+	// Calculate the actuator slew rate due to gyro feedback
+	const float gyro_term_slew_rate = (gyro_term_hpf - _last_gyro_term_hpf) / dt;
+	_last_gyro_term_hpf = gyro_term_hpf;
+
 	/* Apply PI rate controller and store non-limited output */
 	/* FF terms scales with 1/TAS and P,I with 1/IAS^2 */
-	_last_output = _bodyrate_setpoint * _k_ff * ctl_data.scaler +
-		       _rate_error * _k_p * ctl_data.scaler * ctl_data.scaler
-		       + _integrator;
+	_last_output = _bodyrate_setpoint * _k_ff * ctl_data.scaler + p_term * _gain_compression_factor + _integrator;
```

Comment:
Why do you calculate the gain compression after this line and not before?

Reply:
That's a leftover from an earlier prototype. Will fix.